File: RPS.py
import random


def convert(inp):
    randomizer = "RPS"
    return randomizer[inp]


def conv_ascii(contestant):
    if contestant == "P":
        return "📃"
    elif contestant == "R":
        return "🪨"
    elif contestant == "S":
        return "✂️"
    else:
        return " "


def winner(x, y, namer):
    if x == y:
        print("DRAW")

    elif x == 'R' and y == 'P':
        print("CPU Wins")
    elif x == 'R' and y == 'S':
        print(namer + " Wins")

    elif x == 'P' and y == 'R':
        print(namer + " Wins")
    elif x == 'P' and y == 'S':
        print("Computer Wins")

    elif x == 'S' and y == 'R':
        print("Computer Wins")
    elif x == 'S' and y == 'P':
        print(namer + " Wins")


def main():
    print("Welcome to ")
    print('''~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~    
    ~~~~~~~~~~~~~~~~  ROCK, PAPER, SCISSORS! ~~~~~~~~~~~~
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    ''')
    namer = input("What is your name?")

    user = input(namer + " Rock, Paper or Scissors?(Rock(R),Paper(P) or Scissors(S):")
    user.upper()
    cpu = random.randint(0, 2)
    computer = convert(cpu)
    print(conv_ascii(user), " VS ", conv_ascii(computer))
    winner(user, computer, namer)
    print("Thanks for playing")

    print("~~~~~~~~~~~~~~~~~~~~~THANKS FOR USING MY PROGRAM!~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~")


main()

File: RPS.java
import java.util.*;
public class RPS {

    static void winner(String x, String y, String namer){
        if (x.equals(y)){
            System.out.println("DRAW!");
        }
        else if (x.equals("R") && y.equals("P")) {
            System.out.println("CPU Wins!");
        }
        else if (x.equals("R") && y.equals("S")) {
            System.out.println(namer + " Wins!");
        }

        else if (x.equals("P") && y.equals("R")){
            System.out.println(namer + " Wins!");
        }
        else if (x.equals("P") && y.equals("S!")){
            System.out.println("CPU Wins");
        }

        else if (x.equals("S") && y.equals("R!")) {
            System.out.println("CPU Wins");
        }
        else if (x.equals("S") && y.equals("P")){
            System.out.println(namer + " Wins!");
        }
    }



    static String conv_ascii( String contestant) {
        switch (contestant) {
            case "P" -> {
                return "\uD83D\uDCC3";
            }
            case "R" -> {
                return "\uD83E\uDEA8";
            }
            case "S" -> {
                return "✂️";
            }
        }
        return "";
    }


    static String converter(Integer computer){
        String[] randomizer = {"R", "P", "S"};
        return randomizer[computer];
    }
    public static void main(String[] args){

        System.out.println("~~~~~~~~~~~~~~~~~~~ROCK~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
        System.out.println("~~~~~~~~~~~~~~~~~~~~~PAPER~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
        System.out.println("~~~~~~~~~~~~~~~~~~~~~~~~SCISSORS~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");


        Scanner rps = new Scanner(System.in);
        System.out.println("WHat is your name?");
        String namer = rps.nextLine();
        System.out.println(namer + " Rock, Paper or Scissors?(Rock(R),Paper(P) or Scissors(S):");
        String user = rps.nextLine();

        Random rand = new Random();
        int cpu = rand.nextInt((2) + 1);
        String computadora = converter(cpu);
        System.out.println(conv_ascii(user) + " VS "+ conv_ascii(computadora));
        winner(user,computadora, namer);
        System.out.println(namer + " thanks for playing!");

        System.out.println("~~~~~~~~~~~~THANKS FOR USING MY PROGRAM!~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
    }
}
